packages/@creature-chess/server-game/src/player/bot/preparingPhase/actions/utils/getStrategicTypes.ts

```diff
@@ -0,0 +1,70 @@
+import { CreatureType, PieceModel, Card } from "@creature-chess/models";
+import { PlayerState, getAllPieces } from "@creature-chess/gamemode";
+
+const getOwnedPieceTypes = (allPieces: PieceModel[]) => {
+	const pieceTypes: { [type: string]: number } = {};
+	/* eslint-disable guard-for-in */
+	for (const ownedPiece in allPieces) {
+		const ownedType: CreatureType = allPieces[ownedPiece].definition.type;
+		if (pieceTypes[ownedType]) {
+			pieceTypes[ownedType] = pieceTypes[ownedType] + 1;
+		} else {
+			pieceTypes[ownedType] = 1;
+		}
+	}
+	return pieceTypes;
+};
+
+const getStrategicTypes = (state: PlayerState) => {
+	const allTypes = [
+		CreatureType.Earth,
+		CreatureType.Fire,
+		CreatureType.Metal,
+		CreatureType.Water,
+		CreatureType.Wood
+	];
+	const allPieces: PieceModel[] = getAllPieces(state);
+	const pieceTypes = getOwnedPieceTypes(allPieces);
+	const strategicTypes = [];
+	/* eslint-disable guard-for-in */
+	for (const piece in allPieces) {
+		const type: CreatureType = allPieces[piece].definition.type;
+		if (pieceTypes[type]) {
+			pieceTypes[type] = pieceTypes[type] + 1;
+		} else {
+			pieceTypes[type] = 1;
+		}
+	}
+	for (const type in allTypes) {
+		if (!pieceTypes[type]) {
```

Comment:
This line breaks everything

Reply:
Maybe solution is to use list of two lists instead of object to represent ownedPieceTypes?